src/main/scala/org/broadinstitute/dsde/agora/server/business/AclBusiness.scala

```diff
@@ -0,0 +1,122 @@
+package org.broadinstitute.dsde.agora.server.busines
+
+import com.google.api.services.storage.model.{ObjectAccessControl, BucketAccessControl}
+import org.broadinstitute.dsde.agora.server.business.{EntityAuthorizationException, NamespaceAuthorizationException}
+import org.broadinstitute.dsde.agora.server.dataaccess.acls.AgoraPermissions.Manage
+import org.broadinstitute.dsde.agora.server.dataaccess.acls.{AgoraPermissions, AuthorizationProvider}
+import org.broadinstitute.dsde.agora.server.model.{AgoraEntity, AgoraEntityType}
+import org.broadinstitute.dsde.agora.server.dataaccess.acls.gcs.GcsClient._
+import spray.routing.RequestContext
+
+//TODO: Lots to refactor in this class. too much duplication
+class AclBusiness(authorizationProvider: AuthorizationProvider) {
+
+  def listNamespaceAcls(context: RequestContext, entity: AgoraEntity, user: String) = {
+    if (authorizationProvider.namespaceAuthorization(entity, user).canManage) {
+      val bucket = entityToBucket(entity)
+      val bucketAcls = listBucketAcl(bucket).setFields("items(entity,role)").execute()
+
+      bucketAcls.toPrettyString
+
+    } else {
+      throw new NamespaceAuthorizationException(AgoraPermissions(Manage), entity)
+    }
+  }
+
+  def insertNamespaceAcl(_context: RequestContext, entity: AgoraEntity, user: String, acl: BucketAccessControl) = {
+    if (authorizationProvider.namespaceAuthorization(entity, user).canManage) {
+      val bucket = entityToBucket(entity)
+      val bucketAcls = getBucket(bucket).setProjection("full").execute().getAcl
+
+      // Filter any existing acls that reference user to be added.
+      val updatedAcls = filterBucketAclsByEntity(bucketAcls, acl.getEntity)
+
+      updatedAcls.add(acl)
+      patchBucketAcls(bucket, updatedAcls).setFields("acl(entity,role)").execute().toPrettyString
+
+    } else {
+      throw new NamespaceAuthorizationException(AgoraPermissions(Manage), entity)
+    }
+  }
+
+  def deleteNamespaceAcl(_context: RequestContext, entity: AgoraEntity, user: String, acl: BucketAccessControl) = {
+    if (authorizationProvider.namespaceAuthorization(entity, user).canManage) {
+      val bucket = entityToBucket(entity)
+      val bucketAcls = getBucket(bucket).setProjection("full").execute().getAcl
+
+      // Filter any existing acls that reference the user to be deleted.
+      val updatedAcls = filterBucketAclsByEntity(bucketAcls, acl.getEntity)
+
+      patchBucketAcls(bucket, updatedAcls).setFields("acl(entity,role)").execute().toPrettyString
+
+    } else {
+      throw new NamespaceAuthorizationException(AgoraPermissions(Manage), entity)
+    }
+  }
+
+  def filterBucketAclsByEntity(acls: java.util.List[BucketAccessControl], entityToFilter: String): java.util.ArrayList[BucketAccessControl] = {
```

Comment:
for this and the comment below, the google client api returns java ArrayList objects, which do not have filter functionality. The java arrays can be converted to scala arrays, so we can use filter, but they need to be converted back to Java in order to be returned to Google's API. I thought it was cleaner to just write the code as I would in Java instead of converting back and forth. I will add a comment to the code to make it clearer though.